sharding-core/sharding-core-common/src/main/java/org/apache/shardingsphere/core/strategy/algorithm/sharding/RangeShardingAlgorithm.java

```diff
@@ -0,0 +1,128 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.core.strategy.algorithm.sharding;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Splitter;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Range;
+import com.google.common.primitives.Longs;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.shardingsphere.api.sharding.standard.PreciseShardingValue;
+import org.apache.shardingsphere.api.sharding.standard.RangeShardingValue;
+import org.apache.shardingsphere.api.sharding.standard.StandardShardingAlgorithm;
+
+import java.util.Collection;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+/**
+ * Range sharding algorithm.
+ * <p>
+ * Range sharding algorithm is similar to the rule of partition table.
+ * User can specify the range by setting the `range.partition.split.value` parameter.
+ * The `range.partition.split.value` parameter is an ordered list of numbers, separated by commas.
+ * </p>
+ * <p>
+ * For example: If the `range.partition.split.value` parameter is set to `1,5,10`,
+ * the parameter will split all values into four intervals——(-∞, 1), [1,5), [5,10), [10, +∞),
+ * which corresponding to partition_0, partition_1, partition_2, partition_3.
+ * The sharding values will be divided into different partition by its value.
+ * </p>
+ */
+public final class RangeShardingAlgorithm implements StandardShardingAlgorithm<Long> {
+
+    private static final String RANGE_PARTITION_SPLIT_VALUE = "range.partition.split.value";
+
+    private Properties properties = new Properties();
+
+    @Override
+    public String doSharding(final Collection<String> availableTargetNames, final PreciseShardingValue<Long> shardingValue) {
+        Preconditions.checkNotNull(properties.get(RANGE_PARTITION_SPLIT_VALUE), "Range sharding algorithm range partition split value cannot be null.");
+        Map<Integer, Range<Long>> partitionRangeMap = getPartitionRangeMap();
+        for (String each : availableTargetNames) {
+            if (each.endsWith(getPartition(partitionRangeMap, shardingValue.getValue()))) {
+                return each;
+            }
+        }
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Collection<String> doSharding(final Collection<String> availableTargetNames, final RangeShardingValue<Long> shardingValue) {
+        Preconditions.checkNotNull(properties.get(RANGE_PARTITION_SPLIT_VALUE), "Range sharding algorithm range partition split value cannot be null.");
+        Map<Integer, Range<Long>> partitionRangeMap = getPartitionRangeMap();
```

Comment:
for line 61 and 73: 
`doSharding()` method will be called while route value, it may be not efficient to build partitionRangeMap in `doSharding` method, I think it can be used as a static field and  built once while initing instance.

Reply:
Good suggestion, I will optimize this question.

Reply:
> for line 61 and 73:
> `doSharding()` method will be called while route value, it may be not efficient to build partitionRangeMap in `doSharding` method, I think it can be used as a static field and built once while initing instance.

Good suggestion, I will optimize this question.

Reply:
@SanmuWangZJU Thanks. :)

Reply:
> for line 61 and 73:
> `doSharding()` method will be called while route value, it may be not efficient to build partitionRangeMap in `doSharding` method, I think it can be used as a static field and built once while initing instance.

@SanmuWangZJU @tristaZero 
Hi, I have tried to use static field to define `partitionRangeMap`, but I found that it is not suitable. Users may instantiate multiple different range partition rules in the same application instance to correspond to different requirements. Static initialization means same range partition rule. Perhaps the instance field is more suitable.

Reply:
@strongduanmu Got it. Maybe we need a `init()` interface function to initialize some local members so as to avoid calculating some local values like `partitionRangeMap`